src/main/java/com/example/board/service/impl/RankUpStandardImpl.java

```diff
@@ -0,0 +1,147 @@
+package com.example.board.service.impl;
+
+import static com.example.board.domain.type.RankType.ADMIN;
+import static com.example.board.exception.ErrorCode.ALREADY_RANK;
+import static com.example.board.exception.ErrorCode.NOT_ADD_RANK;
+import static com.example.board.exception.ErrorCode.NOT_FIND_RANK;
+import static com.example.board.exception.ErrorCode.OVER_RANK_CHECK_BOARD_COMMENT;
+import static com.example.board.exception.ErrorCode.UNDER_RANK_CHECK_BOARD_COMMENT;
+
+import com.example.board.domain.dto.RankUpStandardDto.AddRankUpStandard;
+import com.example.board.domain.dto.RankUpStandardDto.ModifyRankUpStandard;
+import com.example.board.domain.dto.RankUpStandardDto.SearchRankUpStandard;
+import com.example.board.domain.entity.RankUpStandard;
+import com.example.board.domain.entity.User;
+import com.example.board.domain.repository.RankUpStandardRepository;
+import com.example.board.domain.type.RankType;
+import com.example.board.exception.GlobalException;
+import com.example.board.service.RankUpStandardService;
+import com.example.board.service.UserTypeCheckService;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Service
+@RequiredArgsConstructor
+public class RankUpStandardImpl implements RankUpStandardService {
+
+    private final UserTypeCheckService userTypeCheckService;
+    private final RankUpStandardRepository rankUpStandardRepository;
+
+
+    // 조회
+    public List<SearchRankUpStandard> searchRankUpStandard(Long userId, boolean userTypeCheck) {
+
+        if (userTypeCheck) {
+            userTypeCheckService.userTypeAdmin(userId);
+        }
+
+        List<SearchRankUpStandard> infoRankUpStandards = new ArrayList<>();
+
+        for (RankUpStandard rankUpStandard : rankUpStandardRepository.findAllByOrderByRankName()) {
+            SearchRankUpStandard searchRankUpStandard = SearchRankUpStandard.builder()
+                .standardId(rankUpStandard.getStandardId())
+                .rankName(rankUpStandard.getRankName())
+                .boardCount(rankUpStandard.getBoardCount())
+                .commentCount(rankUpStandard.getCommentCount())
+                .autoRankUpFlag(rankUpStandard.isAutoRankUpFlag())
+                .build();
+
+            infoRankUpStandards.add(searchRankUpStandard);
+        }
+
+        return infoRankUpStandards;
+    }
+
+
+    // 저장
+    @Transactional
+    public List<SearchRankUpStandard> addRankUpStandard(
+        Long userId, AddRankUpStandard addRankUpStandard) {
+
+        if (rankUpStandardRepository.existsByRankName(addRankUpStandard.getRankName())) {
+            throw new GlobalException(ALREADY_RANK);
+        } else if (addRankUpStandard.getRankName().equals(ADMIN)) {
+            throw new GlobalException(NOT_ADD_RANK);
+        }
```

Comment:
서로 다른 예외를 던지고 있는데요. 2개의 if 문이 서로 연관성은 없어보여서 if 문을 따로따로 만드는 것은 어떨까요?

```suggestion
        if (rankUpStandardRepository.existsByRankName(addRankUpStandard.getRankName())) {
            throw new GlobalException(ALREADY_RANK);
        }
        if (addRankUpStandard.getRankName().equals(ADMIN)) {
            throw new GlobalException(NOT_ADD_RANK);
        }
```

Reply:
@spring-kotlin-lover 
넵!!